002-dask-cli.md

```diff
@@ -0,0 +1,79 @@
+**Dask Design Document - 002**
+
+# Dask CLI
+
+**Authors**:
+
+- Jacob Tomlinson (jtomlinson@nvidia.com)
+- Matthew Rocklin
+- Julia Signell
+
+**Created**: 2022-07-20 (Last Updated: 2022-07-20)
+
+
+## Abstract
+
+We propose implementing a `dask` CLI tool that unites the variety of existing CLI tools that are littered across the Dask ecosystem. This CLI will cover new and existing functionality and will be extensible and support dispatching to other CLI tools.
+
+## Motivation
+
+Today we have the follow CLI tools:
+
+- Distributed
+  - `dask-scheduler` - Starts a scheduler
+  - `dask-worker` - Starts a worker from CLI flags
+  - `dask-spec` - Starts a worker from a JSON spec
+  - `dask-ssh` - Launches a Dask cluster on remote machines using SSH
+- Dask Control
+  - `daskctl`
+    - Allows list/scale/delete of Dask clusters that implement the `from_name` method for reconstruction of cluster objects.
+    - Create Dask clusters from a specification file.
+    - Also has an experimental TUI for viewing clusters and also has a CLI dashboard for systems where forwarding web ports is hard.
+- Dask Foo
+  - Many other `dask-foo` projects have a `dask-foo` command that launches a cluster via the CLI. See `dask-yarn` or `dask-ecs`.
+
+This fragmentation leads to user confusion around all of these tools. It also means there are many inconsistencies in how these are implemented. Some use Click for simplified arg parsing, some use Rick for pretty terminal output, etc.
+
+A primary motivation would be to unite everything under one CLI, then a secondary goal would be to improve consistency across tools.
```

Comment:
To go a step further, allowing users easy access to and discoverability of the cluster lifecycle commands in daskctl will enable them to more effectively manage state and deepen understanding. We believe that to be a full-featured production tool, Dask needs to provide tooling that prioritizes non-notebook workflows.